appcues/src/main/java/com/appcues/statemachine/effects/AwaitDismissEffect.kt

```diff
@@ -0,0 +1,21 @@
+package com.appcues.statemachine.effects
+
+import com.appcues.action.ActionProcessor
+import com.appcues.statemachine.Action
+import com.appcues.statemachine.SideEffect
+import kotlinx.coroutines.CompletableDeferred
+import org.jetbrains.annotations.VisibleForTesting
+
+internal data class AwaitDismissEffect(private val action: Action) : SideEffect {
```

Comment:
Renaming AwaitEffect to AwaitDismissEffect since its being used in a particular place (when moving to EndingStep and waiting for UI to dismiss), better naming to help replace a boolean we had describing when we shouldPresent the new container, we know that every time we have this Effect it also means we want to present the next.